### DISCUSSION TITLE
内存超限制问题

### DISCUSSION THREAD

**Discussion by 731879846kk** [apache/doris#15419] [A - General / Q&A] [answered]
最近看到了我们升级到了1.2.0版本，于是将本地虚拟机demo也进行升级进行测试，但是发现了内存超限导致导入失败。查看官网内存超限错误分析  
process memory used 2.68 GB exceed limit 2.47 GB or sys mem available 50.95 GB less than low water mark 3.20 GB, failed alloc size 2.00 MB    
我本地的日志是这样的：
"cancelled: Memory limit exceeded:<consuming tracker:<Load#Id=eb4a363fe74bc7de-9e5a590af8667994>, process memory used 1.51 GB exceed limit 4.38 GB or sys mem available 522.49 MB less than low water mark 550.22 MB, failed alloc size 3.51 MB>, executing msg:<execute:<NodeChannel:indexID=10009:threadId=140183356356352>>. backend 192.168.10.102 process memory used 1.51 GB, limit 4.38 GB. If query tracker exceed, `set exec_mem_limit=8G` to change limit, details see be.INFO.",

我想知道的是，我在1.1.2版本同样是该测试样例，stream_load 导入1G的数据能正常导入，但是1.2.0的这个内存保护机制一直让我无法正常导入，这个信息我还是不理解， 1.51GB 并没有超出4.38的机器内存限制，为什么还是导入不了
我自己有没有什么解决途径呢，比如设置一下导入内存或者什么的？ 虽然这样能避免be被OOM掉，但是我现在也无法导入了...

**Comment by LemonLiTree** [comment, marked as answer]
这个是在1.2版本be加了参数 ： max_sys_mem_available_low_water_mark_bytes ，这个主要是为了Full GC 预留内存的，避免oom，你可以尝试调小这个参数。

**Comment by 731879846kk** [comment]
尝试将这个值调小，目测没有这个问题出现了。 查了一下官网配置，max_sys_mem_available_low_water_mark_bytes 该值在大于16G内存服务器上默认是1.6G， 为了避免内存浪费，走默认或者设置大一些。对于小于16G的机器，比如虚拟机，可以设置小一些
